storage/postgres/postgres.go

```diff
@@ -467,7 +467,12 @@ func (p *Postgres) CreateStore(ctx context.Context, store *openfgapb.Store) (*op
 		return nil, handlePostgresError(err)
 	}
 
-	return store, nil
+	createdStore, err := p.GetStore(ctx, store.Id)
```

Comment:
My preference here is to not call a storage method inside another storage method. I don't think adding the timestamp metadata in the response of `CreateStore` is largely that helpful anyways.

I'm curious if we're ok just returning the `id` and `name` and if the user wants other metadata about the store after it's created, then they can call the `GetStore` RPC. I think simpler is better here.

Reply:
> the only problem I see with this approach is that other storage implementations aren't going to have this ability (such as dynamodb).

It would be the responsibility of the storage implementation to match the spec. 

> That being said, if we're going to return calculated metadata from the storage layer, let's use RETURNING here instead.

Agreed, specially since its PSQL's responsibility to generate it and there is no need for the roundtrip of operations.

Reply:
What I don't like about adding this metadata to the storage layer is that it fundamentally will introduce clock skew issues into the system.

To implement this storage method for Dynamodb we'll have to do something like

```
var store *openfgapb.Store
store.CreatedAt = time.Now()
dynamodb.PutItem(store)

return store
```
The `time.Now` piece isn't a serially ordered timestamps as viewed from the txns against the database. In postgres you can use the [pg_xact_commit_timestamp(xid)](https://www.postgresql.org/docs/current/functions-info.html#FUNCTIONS-COMMIT-TIMESTAMP) to get an accurate txn commit timestamp, but not all databases have that functionality.

This approach just adds complexities that aren't really warranted IMO.

Reply:
After giving it some thought, I agree. It would be nice to simplify our apis at some point.